spec/cipherstash/analysis/text_processor_spec.rb

```diff
@@ -18,6 +18,44 @@
   end
 
   describe "Standard text processor with an ngram filter" do
+    ["1", { foo: "bar" }, Object.new].each do |length|
+      it "raises an error if invalid length of #{length.inspect} provided" do
+        expect {
+          CipherStash::Analysis::TextProcessor.new({
+            "tokenFilters" => [
+              { "kind" => "downcase" },
+              { "kind" => "ngram", "minLength" => length, "maxLength" => length }
+              ],
+            "tokenizer" => { "kind" => "standard" }
+          })
+        }.to raise_error(CipherStash::Client::Error::InternalError, "The values provided to the min and max length must be of type Integer.")
+      end
+    end
+
+    it "raises an error if the min length is greater than the max length" do
```

Comment:
Might be worth adding a test for when min and max length are equal, too

Reply:
@CDThomas I think that's ok for the values to be equal, if for example they want to specify trigrams, they can set the min and max length to be equal.

Reply:
Ah yeah, I meant that it might be worth adding coverage for that scenario (not that it should be an error necessarily). There might already be a case for that and I missed it when I skimmed the existing tests though.

Reply:
oh, got you, i'll add one in!

Reply:
https://github.com/cipherstash/ruby-client/pull/30 @CDThomas quick one to add that additional test.